### DISCUSSION TITLE
Solution for utilizing go-ios on linux without access to xcode for initial WDA installation?

### DISCUSSION THREAD

**Discussion by iDempsey** [danielpaulus/go-ios#126] [Q&A]
Possible dumb question, but is there any solution for being able to utilize go-ios without first connecting the device to a mac for the webdriveragent installation? My goal is to use go-ios for Appium testing on an ipod touch tethered to a linux box but it's my understanding that in order to do this, the connected device requires a webdriveragent installed on it (done through xcode). Any feedback over this is greatly appreciated, thanks!

**Comment by danielpaulus** [comment]
You should be able to use github actions to do that. 
https://github.com/danielpaulus/wda-signer
I have set up a repo here to sign a pre-built wda. Let me know if that works.

**Comment by nanoscopic** [comment]
You'll always need to get a copy of the developer image relevant to your device from a copy of Xcode. In some cases, with recent iOS versions, this means you'll need access to a Beta Xcode, the contents of which cannot be legally shared on Github.

Once you have that proper developer image, you can mount it using go-ios on Linux without involving a mac.

You will also need a Mac developer account to create a provisioning profile ( paid ), or to use Xcode to create a temporary 1 week signing.
